.github/workflows/build.yml

```
@@ -573,6 +573,50 @@ jobs:
         with:
           flags: dotnet
 
+  publish-test-results:
+    name: "Publish Tests Results"
+    needs: test-bicep
+    runs-on: ubuntu-latest
+    permissions:
+      checks: write
+      # only needed unless run with comment_mode: off
+      pull-requests: write
+    if: always()
+
+    steps:
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: Bicep.TestResults.win-x64
+          path: TestResults/win-x64
+
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: Bicep.TestResults.linux-x64
+          path: TestResults/linux-x64
+
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: Bicep.TestResults.linux-musl-x64
+          path: TestResults/linux-musl-x64
+        
+      - name: Download Artifacts
+        uses: actions/download-artifact@v3
+        with:
+          name: Bicep.TestResults.osx-x64
+          path: TestResults/osx-x64
+
+      - name: List downloaded artifacts
+        run: ls -R    
+
+      - name: Publish Test Results
+        uses: EnricoMi/publish-unit-test-result-action@v2
+        with:
+          files: "TestResults/**/*.trx"
```

You get multiple results by running the action multiple times on different files, for instance:

```suggestion
  publish-test-results:
    name: "Publish Tests Results (${{ matrix.rid }})"
    needs: test-bicep
    runs-on: ubuntu-latest
    permissions:
      checks: write
      # only needed unless run with comment_mode: off
      pull-requests: write
    strategy:
      fail-fast: false
      matrix:
        rid: ["win-x64", "linux-x64", "linux-musl-x64", "osx-x64"]
    if: always()

    steps:
      - name: Download Artifacts
        uses: actions/download-artifact@v3
        with:
          name: Bicep.TestResults.${{ matrix.rid }}
          path: TestResults/

      - name: List downloaded artifacts
        run: ls -R TestResults/

      - name: Publish Test Results
        uses: EnricoMi/publish-unit-test-result-action@v2
        with:
          check_name: "Test Results (${{ matrix.rid }})"
          files: "TestResults/**/*.trx"
```